src/Tools/Microsoft.dotnet-openapi/src/Program.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using Microsoft.DotNet.Openapi.Tools;
+
+namespace Microsoft.DotNet.OpenApi
+{
+    public class Program
+    {
+        public static int Main(string[] args)
+        {
+            var outputWriter = new StringWriter();
+            var errorWriter = new StringWriter();
+
+            DebugMode.HandleDebugSwitch(ref args);
+
+            try
+            {
+                using var httpClient = new HttpClientWrapper(new HttpClient());
+                var application = new Application(
+                    Directory.GetCurrentDirectory(),
+                    httpClient,
+                    outputWriter,
+                    errorWriter);
+
+                var result = application.Execute(args);
+
+                return result;
+            }
+            catch(Exception ex)
+            {
+                Console.Error.WriteLine("Unexpected error:");
+                Console.Error.WriteLine(ex.ToString());
+            }
+            finally
+            {
+                var output = outputWriter.ToString();
+                var error = errorWriter.ToString();
+
+                outputWriter.Dispose();
+                errorWriter.Dispose();
+
+                Console.Write(output);
```

Comment:
This would produce out-of-order output when something throws. 

```
Unexpected error: 
Some error
Stack trace...Downloading file from URL blah
```

I'm not sure how valuable regular logs are if something throws. Should this happen inside of the try block? Also, I think it might make sense for these to be `WriteLine` rather than regular `Write` since we're not trying to produce interleaved output.